riptide-compression/src/test/java/org/zalando/riptide/compression/RequestCompressionPluginTest.java

```diff
@@ -0,0 +1,140 @@
+package org.zalando.riptide.compression;
+
+import com.github.restdriver.clientdriver.ClientDriver;
+import org.apache.http.impl.client.HttpClients;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.BufferingClientHttpRequestFactory;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+import org.zalando.riptide.Http;
+import org.zalando.riptide.Plugin;
+import org.zalando.riptide.httpclient.ApacheClientHttpRequestFactory;
+import org.zalando.riptide.httpclient.ApacheClientHttpRequestFactory.Mode;
+
+import java.util.HashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static com.github.restdriver.clientdriver.ClientDriverRequest.Method.POST;
+import static com.github.restdriver.clientdriver.RestClientDriver.giveResponse;
+import static com.github.restdriver.clientdriver.RestClientDriver.onRequestTo;
+import static java.util.Arrays.asList;
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
+import static org.hamcrest.Matchers.emptyString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.springframework.http.HttpHeaders.CONTENT_ENCODING;
+import static org.zalando.riptide.PassRoute.pass;
+
+class RequestCompressionPluginTest {
+
+    private final ClientDriver driver = GzipClientDriver.create();
+    private final ExecutorService executor = newSingleThreadExecutor();
+
+    @AfterEach
+    void tearDown() throws Exception {
+        executor.shutdown();
+        executor.awaitTermination(10, TimeUnit.SECONDS);
+        driver.verify();
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(RequestFactorySource.class)
+    void shouldCompressRequestBody(final ClientHttpRequestFactory factory) {
+        driver.addExpectation(onRequestTo("/")
+                        .withMethod(POST)
+                        .withHeader("X-Content-Encoding", "gzip") // written by Jetty's GzipHandler
```

Comment:
Is Jetty removing the original header and adds this one to indicate it was compressed?

Reply:
Yes, I did not want to put some listener/callback mechanism in my (un-)gzipping rest client driver Jetty hack for now...